sdk/python/kubeflow/training/training_container/hf_llm_training.py

```diff
@@ -0,0 +1,94 @@
+import argparse
+from transformers import (
+    AutoModelForCausalLM,
+    AutoTokenizer,
+    AutoConfig,
+    TrainingArguments,
+    DataCollatorForLanguageModeling,
+    Trainer,
+)
+import torch
+from datasets import load_dataset
+from peft import LoraConfig, get_peft_model
+
+
+def setup_model_and_tokenizer(token_dir, model_dir):
+    # Set up the model and tokenizer
+    tokenizer = AutoTokenizer.from_pretrained(
+        token_dir, use_fast=False, trust_remote_code=True
+    )
+    tokenizer.pad_token = tokenizer.eos_token
+    tokenizer.add_pad_token = True
+
+    model = AutoModelForCausalLM.from_pretrained(
+        model_dir,
+        device_map="auto",
+        trust_remote_code=True,
+    )
+
+    # Freeze model parameters
+    for param in model.parameters():
+        param.requires_grad = False
+
+    return model, tokenizer
+
+
+def load_and_preprocess_data(dataset_dir, tokenizer):
+    # Load and preprocess the dataset
+    train_data = load_dataset(dataset_dir, split="train").map(
+        lambda x: tokenizer(x["text"]), batched=True
+    )
+    train_data = train_data.train_test_split(shuffle=True, test_size=200)
```

Comment:
Why `test_size=200` ?
Should we set test_size=0.1 for 10% split, as suggested here: https://huggingface.co/docs/datasets/process#split